packages/sit-onyx/src/composables/useLenientMaxLengthValidation.ts

```diff
@@ -0,0 +1,111 @@
+import { computed } from "vue";
+import { injectI18n } from "../i18n";
+
+/**
+ * @see [MDN autocapitalize](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autocapitalize)
+ */
+export const AUTOCAPITALIZE = ["none", "sentences", "words", "characters"] as const;
+export type Autocapitalize = (typeof AUTOCAPITALIZE)[number];
+
+/**
+ * Same as TypeScript native "Autofill" type but without "AutoFillSection" because
+ * the Vue compiler currently can not handle it (too complex union type).
+ *
+ * @since TypeScript version 5.2
+ */
+export type Autocomplete = Exclude<AutoFill, AutoFillSection | "">;
+
+/**
+ * Shared types for all kind of text inputs, namely: `<input type="text">` and `<textarea>`
+ */
+export type SharedTextInputProps = {
+  /**
+   * Maximum number of characters that are allowed to be entered.
+   * Warning: when the value is (pre)set programmatically,
+   * the input invalidity will not be detected by the browser, it will only turn invalid
+   * as soon as a user interacts with the input (types something).
+   */
+  maxlength?: number;
+  /**
+   * Restricts the user from typing more characters than allowed.
+   */
+  strictMaxlength?: boolean;
```

Comment:
Since this prop only makes sense when `maxlength` is set, what do you think about combining them into one property like we are already doing for other props/components?

So the user can do:
```vue
<OnyxInput :maxlength="42" />
<OnyxInput :maxlength="{ max: 42, strict: true }" />
```